src/segment-loader.js

```diff
@@ -279,6 +279,13 @@ export default class SegmentLoader extends videojs.EventTarget {
 
         if (segmentInfo && !segmentInfo.isSyncRequest) {
```

Comment:
Why do we only do these adjustments for non-sync requests. It probably doesn't matter if we don't save the info on an outdated segment object when its a sync-request because we don't append it and if we do need to append it in the future, we are going to have to re-request/probe/sync it anyway. However, I think it probably doesn't hurt to make sure the timing info we get from the probe of the sync request to be put onto the segment object. It might help make other calculations that fallback to target duration when start and end times are unavailable be more accurate